python/tchannel/tornado/connection.py

```diff
@@ -554,4 +574,19 @@ def send_request(self, request):
         future = tornado.gen.Future()
         self._outstanding[request.id] = future
         self.post_request(request)
+
+        def adapt_tracing(f):
+            # fetch the request tracing for response
+            f.result().tracing = request.tracing
```

Comment:
Basically, your hooks shouldn't be modifying values like that. If you have a use case where you need to modify a value, that's not a hook. That's part of the main flow of the program.